finesse/hardware/serial_device.py

```diff
@@ -64,7 +61,16 @@ def _get_usb_serial_ports() -> dict[str, str]:
     return _serial_ports
 
 
-class SerialDevice(AbstractDevice):
+class SerialDevice(
+    AbstractDevice,
+    parameters={
+        "port": (
+            "USB port, including vendor ID, product ID and serial number",
+            tuple(_get_usb_serial_ports().keys()),
+        ),
+        "baudrate": ("Baud rate", BAUDRATES),
+    },
+):
```

Comment:
I think you have mentioned this before, but I do not remember the answer:

This will retrieve the available USB ports that have "something" connected to them when this module is loaded, right? So, if the device of interest is off because we forgot to switch it on, then it will not be found. If we switch on the device, the collection of available USB ports will not be updated, so it will not be possible to connect to the device, anyway, the only solution being to restart the software. 

Is this correct or am I misunderstanding anything? Nor that you need to change anything, just checking what the protocol is in that, very common, situation.

Reply:
Correct! This is the same as the existing behaviour and there is already an open issue: #335 

I have an idea of how to fix it: the frontend could send a message to force a refresh of the COM ports before re-requesting the list of plugins. But it's more of a "could have" feature, so I haven't looked at it yet.